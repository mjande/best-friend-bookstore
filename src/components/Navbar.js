import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faPaw, faCaretDown } from "@fortawesome/free-solid-svg-icons";
import { Link } from "react-router-dom";
import "../styles/Navbar.css";

const NavBar = () => {
  return (
    <nav>
      <div className="nav-left">
        <FontAwesomeIcon icon={faPaw} size="xl" />
        <h1>Best Friend Bookstore</h1>
      </div>

      <div className="nav-links">
        <div>Fiction</div>
        <div>Nonfiction</div>
        <div>
          Categories
          <FontAwesomeIcon icon={faCaretDown} />
        </div>
      </div>
    </nav>
  );
};

export default NavBar;
