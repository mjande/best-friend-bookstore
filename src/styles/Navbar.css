nav {
  box-sizing: border-box;

  display: flex;
  justify-content: space-between;
  align-items: flex-end;
  padding: 0 2rem 0.75rem;
  height: 10vh;
  border-bottom: solid;
}

.nav-left,
.nav-links {
  display: flex;
  justify-content: space-between;
  gap: 1rem;
  align-items: flex-end;
}

.cart-link,
.categories-link {
  display: flex;
  align-items: center;
  gap: 0.5rem;
}
