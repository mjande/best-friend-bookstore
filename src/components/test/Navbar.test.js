import React from "react";
import { render, screen } from "@testing-library/react";
import Navbar from "../Navbar";
import { BrowserRouter } from "react-router-dom";

describe("Navbar", () => {
  it("renders navbar with header and links", () => {
    const { container } = render(<Navbar />, { wrapper: BrowserRouter });

    const headerElement = screen.getByRole("heading");
    const links = screen.getAllByRole("link");

    expect(headerElement.textContent).toBe("Booklovers Bookstore");
    expect(links).toHaveLength(9);

    expect(container).toMatchSnapshot();
  });
});
