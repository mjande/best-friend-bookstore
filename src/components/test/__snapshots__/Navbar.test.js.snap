// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders navbar with header and links 1`] = `
<div>
  <nav>
    <div
      class="nav-left"
    >
      <a
        class="header"
        href="/"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-book fa-xl "
          data-icon="book"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M96 0C43 0 0 43 0 96V416c0 53 43 96 96 96H384h32c17.7 0 32-14.3 32-32s-14.3-32-32-32V384c17.7 0 32-14.3 32-32V32c0-17.7-14.3-32-32-32H384 96zm0 384H352v64H96c-17.7 0-32-14.3-32-32s14.3-32 32-32zm32-240c0-8.8 7.2-16 16-16H336c8.8 0 16 7.2 16 16s-7.2 16-16 16H144c-8.8 0-16-7.2-16-16zm16 48H336c8.8 0 16 7.2 16 16s-7.2 16-16 16H144c-8.8 0-16-7.2-16-16s7.2-16 16-16z"
            fill="currentColor"
          />
        </svg>
        <h1>
          Booklovers Bookstore
        </h1>
      </a>
      <div
        class="dropdown"
      >
        <div
          aria-hidden="true"
          class="dropdown-link"
        >
          Categories
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-caret-down "
            data-icon="caret-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M137.4 374.6c12.5 12.5 32.8 12.5 45.3 0l128-128c9.2-9.2 11.9-22.9 6.9-34.9s-16.6-19.8-29.6-19.8L32 192c-12.9 0-24.6 7.8-29.6 19.8s-2.2 25.7 6.9 34.9l128 128z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          class="dropdown-content"
        >
          <a
            href="/books/subject/mystery"
          >
            Mystery
          </a>
          <a
            href="/books/subject/romance"
          >
            Romance
          </a>
          <a
            href="/books/subject/science-fiction"
          >
            Science Fiction
          </a>
          <a
            href="/books/subject/fantasy"
          >
            Fantasy
          </a>
          <a
            href="/books/subject/thriller"
          >
            Thriller
          </a>
          <a
            href="/books/subject/memoir"
          >
            Memoir
          </a>
          <a
            href="/books/subject/essays"
          >
            Essays
          </a>
        </div>
      </div>
      <form>
        <input
          name="search"
          type="text"
        />
        <button>
          Search
        </button>
      </form>
    </div>
    <div
      class="nav-right"
    >
      <a
        class="cart-link"
        href="/cart"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-cart-shopping "
          data-icon="cart-shopping"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 24C0 10.7 10.7 0 24 0H69.5c22 0 41.5 12.8 50.6 32h411c26.3 0 45.5 25 38.6 50.4l-41 152.3c-8.5 31.4-37 53.3-69.5 53.3H170.7l5.4 28.5c2.2 11.3 12.1 19.5 23.6 19.5H488c13.3 0 24 10.7 24 24s-10.7 24-24 24H199.7c-34.6 0-64.3-24.6-70.7-58.5L77.4 54.5c-.7-3.8-4-6.5-7.9-6.5H24C10.7 48 0 37.3 0 24zM128 464a48 48 0 1 1 96 0 48 48 0 1 1 -96 0zm336-48a48 48 0 1 1 0 96 48 48 0 1 1 0-96z"
            fill="currentColor"
          />
        </svg>
        <div
          data-testid="cart-items-count"
        >
           items
        </div>
      </a>
    </div>
  </nav>
</div>
`;
